src/liboslexec/wide/wide_opclosure.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright Contributors to the Open Shading Language project.
+// SPDX-License-Identifier: BSD-3-Clause
+// https://github.com/AcademySoftwareFoundation/OpenShadingLanguage
+
+
+#include <iostream>
+#include <cmath>
+
+#include <OSL/genclosure.h>
+#include <OSL/batched_shaderglobals.h>
+
+#include "oslexec_pvt.h"
+
+#include "define_opname_macros.h"
+
+OSL_NAMESPACE_ENTER
+namespace __OSL_WIDE_PVT {
+
+OSL_USING_DATA_WIDTH(__OSL_WIDTH)
+
+
+OSL_BATCHOP void
+__OSL_MASKED_OP(add_closure_closure) (void *bsg_, void *wide_out_, 
+        void *wide_closure_a_, void* wide_closure_b_, unsigned int mask_value)
+{
+    // TODO avoid allot if !a or !b?
+    auto *bsg = reinterpret_cast<BatchedShaderGlobals *>(bsg_);
+    Wide<ClosureColorPtr> wide_out (wide_out_);
+    Wide<ClosureColorPtr> wide_closure_a (wide_closure_a_);
+    Wide<ClosureColorPtr> wide_closure_b (wide_closure_b_);
+    Mask mask (mask_value);
+    
+    ClosureAdd *add = bsg->uniform.context->batched<__OSL_WIDTH>().closure_add_allot ();
```

Comment:
Is *add a flat array of pointers?  Is it aligned?  Then could probably make it
`Wide<ClosureAdd *> add(bsg->uniform.context->batched<__OSL_WIDTH>().closure_add_allot ());`